src/nock.ts

```diff
@@ -109,6 +113,29 @@ function getPathRegex(pathTemplate: string): RegExp {
   return new RegExp(escapedPath.replace(/ ([^ ]+) /g, '([^?/]+)'));
 }
 
+function normalizeHeaderValue(value: unknown): string {
+  if (Array.isArray(value)) {
+    assert(value.length === 1, 'unexpected length of header array');
```

Comment:
We have one example where the `content-type` header is being an array with a single item. This new code would try to normalize all of the headers, so it's kinda possible that we have some random headers being an array with more than two items.. no?

Reply:
yes. However I do not know what are the semantics there so assert that this does not happen and fix when somebody trips to that

Reply:
💪